_posts/2025-08-26-strimzi-metrics-reporter.md

```diff
@@ -0,0 +1,89 @@
+Up to this point, we have recommended using the [JMX Exporter](https://github.com/prometheus/jmx_exporter) as a way to expose Kafka metrics, and we have provided some example YAML files in our [Strimzi Examples folder](https://github.com/strimzi/strimzi-kafka-operator/tree/main/packaging/examples/metrics).
+
+The [Strimzi Metrics Reporter](https://github.com/strimzi/metrics-reporter) is a new project in Strimzi, still in early access, and it is available since Strimzi 0.48.0 release.
+It implements the Kafka `MetricsReporter` interface to expose metrics via Prometheus.
+Strimzi Metrics Reporter is designed from scratch as a Kafka plugin and allows you to configure it directly from the Kafka configuration.
+That also means it is more efficient compared to other monitoring tools that run as sidecars or JVM agents.
+Strimzi Metrics Reporter has fixed names for all metrics giving users much less flexibility than other tools which provide complex mapping rules based on regular expressions.
+But it is easier to use and provides a metrics interface that is less fragile and fragmented, and more like an API.
+
+Here, we will discuss Strimzi Metrics Reporter configuration, and guide you on how to use it effectively to monitor your Kafka clusters.
+
+### Key Features
+* Native Prometheus support: The reporter exposes metrics in the Prometheus format through an HTTP endpoint, without relying on JMX.
+* Configurable Metrics Collection: Allows users to specify which metrics should be exposed using a flexible allowlist.
+
+#### Deploying Metrics Reporter
+The first step is to include the Strimzi Metrics Reporter in your `Kafka` custom resource.
+To do this, add the following:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+  # ...
+    metricsConfig:
+      type: strimziMetricsReporter
+  # ...  
+```
+
+By adding `type: strimziMetricsReporter` to the `metricsConfig` section of your `Kafka` custom resource, Strimzi will export a sensible set of default metrics.
+However, you can add you own custom values by filtering the metrics by name. 
+This is achieved by adding another field called `values`, and within that field, adding `allowList`.
+Then you can specify which metrics you want to collect, where each entry is used to filter allowed metrics.
+You can put individual metrics names or use a regex to include a group of metrics and avoid having a long list.
+For example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+  # ...
+    metricsConfig:
+      type: strimziMetricsReporter
+      values:
+        allowList:
+          - "kafka_log.*"
+          - "kafka_network.*"
+  # ...  
+```
+
+If you are adding Strimzi Metrics Reporter to an existing cluster, then you will see all your Kafka broker and Controller pods roll so that the Strimzi Metrics Reporter config is set in your `Kafa` resource.
```

Comment:
```suggestion
If you are adding Strimzi Metrics Reporter to an existing cluster, then you will see all your Kafka broker and Controller pods roll so that the Strimzi Metrics Reporter config is set in your `Kafka` resource.
```